### PR TITLE
Adding extended hostname capability in sys_info.rb for SolarisOS 11

### LINKED ISSUE #4850: Solaris support in the sys.info resource
# Summary
Currently inspec is unable to retrieve hostname from sys.info resource

# Describe the problem
sys.info is unable to retrieve hostsname. The test code I'm using is below:
```ruby
describe etc_hosts(hostfile).where {ip_address == sys_info.ip_address } do
  its('primary_name') { should match sys_info.hostname }
  its('all_host_names') { should eq [[sys_info.hostname], [sys_info.fqdn ]] }
end
```
Currenty returns:
resource is not supported with that option on your OS.

# Expected outcome.
sys_info.hostname returns hostname and sys_info.fqdn returns system FQDN. sys_info.ip_address returns system IP address.

# Possible Solution
solaris 11 hostname command returns the FQDN of the system. so split with [0] will be short and [1] will be domain. Will submit a PR from this idea.

### PR DESCRIPTION
**Pull request by n3mawashi**
## Description
Allow inspec to test extended hostname information in sys_info.rb in SolarisOS 11. Added short, long/fqdn and IP along with update to unit testing and mockfile.

## Related Issue
Fixes https://github.com/inspec/inspec/issues/4850

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ X ] New content (non-breaking change)
- [ ] Breaking change (a content change which would break existing functionality or processes)

## Checklist:
- [ X ] I have read the **CONTRIBUTING** document.

### CODE REVIEW

File: lib/inspec/resources/sys_info.rb [hunk 1]
```diff
@@ -84,6 +86,36 @@ def mac_hostname(opt = nil)
...
+      elsif opt == " -s"
+        inspec.command("hostname").stdout.chomp.split(".", 2).first
+      elsif opt == " -I"
+        inspec.command("getent hosts `hostname`").stdout.chomp.split("\t").first
+      else
+        # To allow mock tests to work
+        inspec.command("hostname").stdout.chomp
+      end
+    end
     # returns the Manufacturer of the local system
```
Comment (clintoncwolfe) [inline lib/inspec/resources/sys_info.rb:120]: The linter is complaining about the lack of a blank line here - you can add it manually, or check for linter issues using `bundle exec chefstyle` . Add `-a` to have them auto-corrected.

Comment (n3mawashi) [reply lib/inspec/resources/sys_info.rb:120]: I'm adding it manually.  Thanks for the heads up

Comment (n3mawashi) [reply lib/inspec/resources/sys_info.rb:120]: What's the best way to add files or commits to PR?

Comment (n3mawashi) [reply lib/inspec/resources/sys_info.rb:120]: Resolved

**Comment by james-stocks** [comment]
Hello @ptiggerdine -
Apologies for the lack of response on this. This PR looks good to me, thank you! 
There are a couple of non-functional things to fix before merging:
 * the 9 commits in this PR should be flattened into a single commit
 * there is one linting issue - `lib/inspec/resources/sys_info.rb:119:1: C: Layout/TrailingWhitespace: Trailing whitespace detected.`
 * Github reports that there is now a conflict with other changes to `lib/inspec/resources/etc_hosts.rb` - the PR branch needs rebased against master.

Let us know (here in this PR or on the Community slack) if we can help

**Pull request closed without merging**